internal/service/finspace/kx_cluster_test.go

```diff
@@ -58,6 +59,72 @@ func TestAccFinSpaceKxCluster_basic(t *testing.T) {
 	})
 }
 
+func TestAccFinSpaceKxCluster_invalidEnvironmentId(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping long-running test in short mode")
+	}
+
+	ctx := acctest.Context(t)
+	var kxEnvironment finspace.GetKxEnvironmentOutput
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	invalidEnvironmentId := sdkacctest.RandString(22)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(ctx, t)
+			acctest.PreCheckPartitionHasService(t, finspace.ServiceID)
+		},
+		ErrorCheck:               acctest.ErrorCheck(t, finspace.ServiceID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckKxClusterDestroy(ctx),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKxClusterConfigBase(rName),
+				Check:  testAccCheckKxEnvironmentExists(ctx, "aws_finspace_kx_environment.test", &kxEnvironment),
+			},
+			{
+				Config:      testAccKxClusterConfig_environmentId(rName, invalidEnvironmentId),
+				ExpectError: regexp.MustCompile("ResourceNotFoundException: No matching Environment found with requested ID"),
+			},
+		},
+	})
+}
+
+func TestAccFinSpaceKxCluster_delete(t *testing.T) {
```

Comment:
Is this test needed? testAccCheckKxClusterDestroy(ctx) is called at the end of every test (CheckDestroy field). Could potentially have another disappears test for an RDB like the one we have for an HDB directly below this, but even that might not be necessary really

Reply:
That's a good point. This test is quite redundant, I'll remove it.